File: Selenium/Activities/Activity14.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity14 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();

        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println(driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("/html/body/div/div[2]/div/div[2]/div[2]/table/tbody/tr"));
        System.out.println("Num of rows is : " + rows.size());

        List<WebElement> columns = driver.findElements(By.xpath("/html/body/div/div[2]/div/div[2]/div[2]/table/tbody/tr[1]/td"));
        System.out.println("Num of rows is : " + columns.size());


        WebElement second_row_column = driver.findElement(By.xpath("//*[@id=\"sortableTable\"]/tbody/tr[2]/td[2]"));
        System.out.println(second_row_column.getText());

        WebElement y = driver.findElement(By.xpath("//*[@id=\"sortableTable\"]/thead/tr/th[1]"));
        System.out.println(y.getText());
        y.click();

        WebElement second_row_column1 = driver.findElement(By.xpath("//*[@id=\"sortableTable\"]/tbody/tr[2]/td[2]"));
        System.out.println(second_row_column1.getText());

        List<WebElement> footer = driver.findElements(By.xpath("//*[@id=\"sortableTable\"]/tfoot/tr/th"));
        for (WebElement x : footer){
            System.out.println(x.getText());
        }

        driver.quit();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()
driver.get("https://v1.training-support.net/selenium/tables")
print(driver.title)

rows = driver.find_elements(By.XPATH, '//*[@id="sortableTable"]/tbody/tr')
print(len(rows))

columns = driver.find_elements(By.XPATH, '//*[@id="sortableTable"]/thead/tr/th')
print(len(columns))

second_row_column = driver.find_element(By.XPATH, '//*[@id="sortableTable"]/tbody/tr[2]/td[2]')
print(second_row_column.text)

driver.find_element(By.XPATH, '//*[@id="sortableTable"]/thead/tr[1]/th[1]').click()

second_row_column = driver.find_element(By.XPATH, '//*[@id="sortableTable"]/tbody/tr[2]/td[2]')
print(second_row_column.text)

footer = driver.find_elements(By.XPATH, '//*[@id="sortableTable"]/tfoot/tr/th')
for i in footer:
    print(i.text)

driver.quit()